### ISSUE TITLE
error for name mangled attributes

### ISSUE THREAD

**Issue by ghassemi-ehsan** [microsoft/pyright#750]
hi.
i think pyright cannot handle name mangled attributes.
i am a newbie in python and learning it as  a new language.
have i missed something??

**Comment by erictraut** [comment]
Can you provide a code sample?

**Comment by ghassemi-ehsan** [comment]
```
class Car:
    def __init__(self):
        self.__x = "y"

    __a = "b"


car1 = Car()
print(car1._Car__x)  #  Cannot access member "_Car__x" for type "Car"--Member "_Car__x" is unknown
print(car1.__x)

print(car1._Car__a) # Cannot access member "_Car__a" for type "Car"--Member "_Car__a" is unknown
print(car1.__a)

```

in above sample code i get two errors from pyright.
i wrote error messages as comment in front of error lines...

**Comment by erictraut** [comment]
This is intended behavior. A double underscore in Python means that the attribute is private to the class that defines the attribute. It should not be accessible outside of that class. The type checker enforces this rule. If you want to expose such a member, I recommend defining a property that returns it.

**Comment by ghassemi-ehsan** [comment]
I knew that. I just wondered why type checker shows error for a statement that interpreter executes it with no error, but doesn't show error for next line that is led to an AttributeError.
at the end thanks for your detailed answer and sorry for my bad English.

**Comment by erictraut** [comment]
Ah, thanks for the additional context. (Your English is great, by the way.)

Pyright doesn't report errors for the next line because you haven't enabled that diagnostic rule. By default, the "reportPrivateUsage" rule is disabled. You can enable it for your entire project in a pyrightconfig.json file. Or you can enable it within a specific file by adding the header comment:
```
# pyright: reportPrivateUsage=true
```

Or if you want to enable all diagnostic rules, you can enable "strict" mode:
```
# pyright: strict=true
```

Accessing mangled names directly in your code is flagged as an error always because no one should be doing that. The way in which names are mangled is an implementation detail, so you shouldn't be relying on it in your code. 

There are many things you can do in Python that do not result in a runtime exception but are still considered programming errors. This is a good example. :)
